src/main/java/org/sopt/repository/MemoryMemberRepository.java

```diff
@@ -0,0 +1,37 @@
+package org.sopt.repository;
+
+import org.sopt.domain.Member;
+
+import java.util.*;
+
+public class MemoryMemberRepository {
+
+
+    private static final Map<Long, Member> store = new HashMap<>();
+
+
+    public Member save(Member member) {
+
+        store.put(member.getId(), member);
+        return member;
+
+    }
+
+    public Optional<Member> findById(Long id) {
+        return Optional.ofNullable(store.get(id));
+    }
+
+    public Optional<Member> findByEmail(String email) {
+        return store.values().stream()
+                .filter(member -> member.getEmail().equals(email))
```

Comment:
stream 적용하신 것 좋습니다 👍🏻👍🏻
member에서 email을 직접 가져와서 비교하는 것보다 아래처럼 email을 member에 던져주고 직접 member 객체가 판단하도록 하면 조금 더 객체지향적인 코드가 될 것 같아요!
```suggestion
// Member 클래스에 이메일 비교하는 메서드 생성
public class Member {
    ...
    public boolean hasEmail(String email) {
         return this.email().equals(email);
    }

}

// 요 부분에서는 이렇게 호출하기
.filter(member -> member.hasEmail(email))
```